src/System.Runtime/tests/System/Version.cs

```diff
@@ -3,211 +3,263 @@
-    public void TestMajorMinorConstructor()
+    public static void TestCtor_Int_Int_Invalid()
     {
-        VerifyConstructor(0, 0);
-        VerifyConstructor(2, 3);
-        VerifyConstructor(int.MaxValue, int.MaxValue);
-        Assert.Throws<ArgumentOutOfRangeException>(() => new Version(-1, 0));
-        Assert.Throws<ArgumentOutOfRangeException>(() => new Version(0, -1));
+        Assert.Throws<ArgumentOutOfRangeException>("major", () => new Version(-1, 0)); // Major < 0
+        Assert.Throws<ArgumentOutOfRangeException>("minor", () => new Version(0, -1)); // Minor < 0
     }
 
-    private static void VerifyConstructor(int major, int minor)
+    [Theory]
+    [InlineData(0, 0, 0)]
+    [InlineData(2, 3, 4)]
+    [InlineData(int.MaxValue, int.MaxValue, int.MaxValue)]
+    public static void TestCtor_Int_Int_Int(int major, int minor, int build)
     {
-        var version = new Version(major, minor);
-        VerifyVersion(version, major, minor, -1, -1);
+        VerifyVersion(new Version(major, minor, build), major, minor, build, -1);
     }
 
     [Fact]
-    public void TestMajorMinorBuildConstructor()
+    public static void TestCtor_Int_Int_Int_Invalid()
     {
-        VerifyConstructor(0, 0, 0);
-        VerifyConstructor(2, 3, 4);
-        VerifyConstructor(int.MaxValue, int.MaxValue, int.MaxValue);
-        Assert.Throws<ArgumentOutOfRangeException>(() => new Version(-1, 0, 0));
-        Assert.Throws<ArgumentOutOfRangeException>(() => new Version(0, -1, 0));
-        Assert.Throws<ArgumentOutOfRangeException>(() => new Version(0, 0, -1));
+        Assert.Throws<ArgumentOutOfRangeException>("major", () => new Version(-1, 0, 0)); // Major < 0
+        Assert.Throws<ArgumentOutOfRangeException>("minor", () => new Version(0, -1, 0)); // Minor < 0
+        Assert.Throws<ArgumentOutOfRangeException>("build", () => new Version(0, 0, -1)); // Build < 0
     }
 
-    private static void VerifyConstructor(int major, int minor, int build)
+    [Theory]
+    [InlineData(0, 0, 0, 0)]
+    [InlineData(2, 3, 4, 7)]
+    [InlineData(int.MaxValue, int.MaxValue, int.MaxValue, int.MaxValue)]
+    public static void TestCtor_Int_Int_Int_Int(int major, int minor, int build, int revision)
     {
-        var version = new Version(major, minor, build);
-        VerifyVersion(version, major, minor, build, -1);
+        VerifyVersion(new Version(major, minor, build, revision), major, minor, build, revision);
     }
 
     [Fact]
-    public void TestMajorMinorBuildRevisionConstructor()
+    public static void TestCtor_Int_Int_Int_Int_Invalid()
     {
-        VerifyConstructor(0, 0, 0, 0);
-        VerifyConstructor(2, 3, 4, 7);
-        VerifyConstructor(2, 3, 4, 65535);
-        VerifyConstructor(2, 3, 4, 65536);
-        VerifyConstructor(2, 3, 4, 32767);
-        VerifyConstructor(2, 3, 4, 32768);
-        VerifyConstructor(2, 3, 4, 2147483647);
-        VerifyConstructor(2, 3, 4, 2147450879);
-        VerifyConstructor(2, 3, 4, 2147418112);
```

Comment:
I'm not following.  All I see those testing is int.MaxValue, not these other values (like short.MaxValue, ushort.MaxValue, etc.)... where are those?